dlfcn.c

```diff
@@ -474,6 +474,48 @@ char *dlerror( void )
     return error_buffer;
 }
 
+int dladdr (const void *addr, Dl_info *info)
+{
+   // only returns filename, FWIW.
+   TCHAR  tpath[PATH_MAX];
+   MEMORY_BASIC_INFORMATION mbi;
+   char  *path;
+   char  *tmp;
+   size_t length;
+   int    ret = 0;
+
+   if (!info)
+     return 0;
+
+   ret = GetModuleFileName(NULL, (LPTSTR)&tpath, PATH_MAX);
+   if (!ret)
+     return 0;
+
+   path = tpath;
+
+   length = strlen (path);
+   if (length >= PATH_MAX)
+     {
+       length = PATH_MAX - 1;
+       path[PATH_MAX - 1] = '\0';
+     }
+
+   /* replace '/' by '\' */
+   tmp = path;
+   while (*tmp)
+     {
+        if (*tmp == '/') *tmp = '\\';
+        tmp++;
+     }
+
+   memcpy (info->dli_fname, path, length + 1);

```

Comment:
Whole allocates memory for `info->dli_fname`? And where is specified allocated length?

In `struct dl_info` you have defined `dli_fname` as `const char *dli_fname;` therefore you cannot write to `dli_fname` member via `memcpy`.

This introduce problem with usage of incompatible pointer.

Reply:
struct dl_info has been changed to fix this issue.